contracts/mainnet/connectors/morpho-compound/main.sol

```diff
@@ -0,0 +1,440 @@
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		morphoCompound.borrow(_poolTokenAddress, _amt);
+
+		if (_isETH) convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogBorrow(address,address,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Borrow ETH/ERC20_Token.
+	 * @notice Borrow a token from Morpho Compound with max gas.
+	 * @param _tokenAddress The address of underlying token to borrow.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to borrow.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to borrow.
+	 * @param _maxGasForMatching The maximum amount of gas to consume within a matching engine loop.
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens borrowed.
+	 */
+	function borrowWithMaxGas(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _maxGasForMatching,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+
+		bool _isETH = _tokenAddress == ethAddr;
+
+		morphoCompound.borrow(_poolTokenAddress, _amt, _maxGasForMatching);
+
+		if (_isETH) convertWethToEth(_isETH, TokenInterface(wethAddr), _amt);
+
+		setUint(_setId, _amt);
+
+		_eventName = "LogBorrowWithMaxGas(address,address,uint256,uint256,uint256,uint256)";
+		_eventParam = abi.encode(
+			_tokenAddress,
+			_poolTokenAddress,
+			_amt,
+			_maxGasForMatching,
+			_getId,
+			_setId
+		);
+	}
+
+	/**
+	 * @dev Withdraw ETH/ERC20_Token.
+	 * @notice Withdraw a token from Morpho Compound.
+	 * @param _tokenAddress The address of underlying token to withdraw.(For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param _poolTokenAddress The address of cToken to withdraw.(For ETH: cWETH address)
+	 * @param _amount The amount of the token (in underlying) to withdraw. (For max: `uint256(-1)`)
+	 * @param _getId ID to retrieve amt.
+	 * @param _setId ID stores the amount of tokens withdrawed.
+	 */
+	function withdraw(
+		address _tokenAddress,
+		address _poolTokenAddress,
+		uint256 _amount,
+		uint256 _getId,
+		uint256 _setId
+	)
+		external
+		payable
+		returns (string memory _eventName, bytes memory _eventParam)
+	{
+		uint256 _amt = getUint(_getId, _amount);
+		bool _isETH = _tokenAddress == ethAddr;
+		address _token = _isETH ? wethAddr : _tokenAddress;
+
+		if (_amt == uint256(-1)) {
+			(, , _amt) = morphoCompoundLens.getCurrentSupplyBalanceInOf(
+				_poolTokenAddress,
+				address(this)
+			);
+		}
+
+		morphoCompound.withdraw(_poolTokenAddress, _amt);
```

Comment:
`_amt` is minimumed to the current supply balance of the user withdrawing (https://etherscan.io/address/0x082Bf6702E718483c85423bD279088C215a21302#code#F1#L409) so you can just forward `_amt` to `morphoCompound.withdraw` in case it's too large

Reply:
Yes but they need it to log the correct amount is their event I think

Reply:
Hi @Rubilmax @MerlinEgalite 
Yes, we need `_amt` to log and also to set the ID. In many places, we add the spell amounts directly using `setId` and `getId` , for eg. in refinancing. So, without the amount, that wouldn't be possible.